src/nft/hooks/useFetchAssets.ts

```diff
@@ -0,0 +1,177 @@
+
+  const resetStateBeforeFetch = useCallback(() => {
+    didOpenUnavailableAssets && setDidOpenUnavailableAssets(false)
+    !bagIsLocked && setBagLocked(true)
+    setBagStatus(BagStatus.FETCHING_ROUTE)
+  }, [bagIsLocked, didOpenUnavailableAssets, setBagLocked, setBagStatus, setDidOpenUnavailableAssets])
+
+  return useCallback(async () => {
+    resetStateBeforeFetch()
+
+    if (usingGqlRouting) {
+      fetchGqlRoute({
+        variables: {
+          senderAddress: account ? account : '',
+          nftTrades: buildNftTradeInputFromBagItems(itemsInBag),
+          tokenTrades: tokenTradeInput ? tokenTradeInput : undefined,
+        },
+        onCompleted: (data) => {
+          if (!data.nftRoute || !data.nftRoute.route) {
+            setBagStatus(BagStatus.ADDING_TO_BAG)
+            setBagLocked(false)
+            return
+          }
+
+          const wishAssetsToBuy = getPurchasableAssets(itemsInBag)
+          const purchasingWithErc20 = !!tokenTradeInput
+          const { route, routeResponse } = buildRouteResponse(data.nftRoute, purchasingWithErc20)
+
+          const { newBagItems, nextBagStatus } = getNextBagState(wishAssetsToBuy, route, purchasingWithErc20)
+
+          setItemsInBag(newBagItems)
+          setBagStatus(nextBagStatus)
+
+          if (nextBagStatus === BagStatus.CONFIRMING_IN_WALLET) {
+            purchaseAssets(routeResponse, wishAssetsToBuy, purchasingWithErc20)
+            setBagLocked(true)
+            return
+          }
+
+          setBagLocked(false)
+        },
+      })
+    } else {
+      try {
+        const assetsToBuy = getPurchasableAssets(itemsInBag)
+        const ethSellObject = buildSellObject(
+          assetsToBuy
+            .reduce((ethTotal, asset) => ethTotal.add(BigNumber.from(asset.priceInfo.ETHPrice)), BigNumber.from(0))
+            .toString()
+        )
+        const routeData = await queryClient.fetchQuery(['assetsRoute', ethSellObject, assetsToBuy, account], () =>
+          fetchRoute({
+            toSell: [ethSellObject],
+            toBuy: assetsToBuy,
+            senderAddress: account ?? '',
+          })
+        )
+
+        const { updatedAssets } = compareAssetsWithTransactionRoute(assetsToBuy, routeData.route)
+
+        const fetchedPriceChangedAssets = updatedAssets
+          .filter((asset) => asset.updatedPriceInfo)
+          .sort(sortUpdatedAssets)
+        const fetchedUnavailableAssets = updatedAssets.filter((asset) => asset.isUnavailable)
+        const fetchedUnchangedAssets = updatedAssets.filter((asset) => !asset.updatedPriceInfo && !asset.isUnavailable)
+        const hasReviewedAssets = fetchedUnchangedAssets.length > 0
+        const hasAssetsInReview = fetchedPriceChangedAssets.length > 0
+        const hasUnavailableAssets = fetchedUnavailableAssets.length > 0
+        const hasAssets = hasReviewedAssets || hasAssetsInReview || hasUnavailableAssets
+        const shouldReview = hasAssetsInReview || hasUnavailableAssets
```

Comment:
could we write a util that takes in `updatedAssets` or even `routeData and `assetsToBuy` and returns these params?

Reply:
this is in non feature flag code